paddle/fluid/operators/distributed/request_handler_impl.cc

```diff
@@ -42,13 +43,16 @@ bool RequestSendHandler::Handle(const std::string& varname,
   // Async
   if (!sync_mode_) {
     rpc_server_->Profiler().OneStep();
-    try {
-      executor_->RunPreparedContext((*grad_to_prepared_ctx_)[varname].get(),
-                                    scope);
-    } catch (std::exception& e) {
-      LOG(ERROR) << "async: run sub program error " << e.what();
-      return false;
-    }
+    std::future<void> ft = framework::Async([this, &scope, varname]() {
```

Comment:
Maybe add one comment that why need to run in thread here.

Reply:
done